browser/brave_content_browser_client.cc

```diff
@@ -198,24 +199,25 @@ bool HandleURLRewrite(GURL* url, content::BrowserContext* browser_context) {
   return false;
 }
 
-void BindCosmeticFiltersResources(
+void BindCosmeticFiltersResourcesOnTaskRunner(
     content::RenderFrameHost* const frame_host,
     mojo::PendingReceiver<cosmetic_filters::mojom::CosmeticFiltersResources>
         receiver) {
-  auto* web_contents = content::WebContents::FromRenderFrameHost(frame_host);
-  if (!web_contents)
-    return;
-
-  auto* profile =
-      Profile::FromBrowserContext(web_contents->GetBrowserContext());
-  auto* settings_map = HostContentSettingsMapFactory::GetForProfile(profile);
-
   mojo::MakeSelfOwnedReceiver(
       std::make_unique<cosmetic_filters::CosmeticFiltersResources>(
-          settings_map, g_brave_browser_process->ad_block_service()),
+          g_brave_browser_process->ad_block_service()),
       std::move(receiver));
 }
 
+void BindCosmeticFiltersResources(
+    content::RenderFrameHost* const frame_host,
```

Comment:
frame_host is no longer needed

Reply:
I don't think it can be removed here; this function is added into `BinderMapWithContext` so it needs to have the right signature.

Reply:
ah, got it. Ok